### PR TITLE
Add SpeechTransformer model

### PR DESCRIPTION
**Pull request by xuzhao9**
Solves issue#289. The model supports only CUDA on eager mode.

Runtime information:
```
$ python run.py speech_transformer -d cuda -m eager -t train
Epoch 1 | Iter 1 | Average Loss 7.286 | Current Loss 7.286260 | 840.0 ms/batch
Epoch 1 | Iter 11 | Average Loss 7.230 | Current Loss 6.541576 | 325.4 ms/batch
Epoch 1 | Iter 1 | Average Loss 6.233 | Current Loss 6.232590 | 361.6 ms/batch
Epoch 1 | Iter 11 | Average Loss 6.215 | Current Loss 6.191328 | 123.4 ms/batch
Ran in 5.307560443878174 seconds.

$ python run.py speech_transformer -d cuda -m eager -t eval
Ran in 3.965590476989746 seconds.
```

I added four files: `__init__.py`, `config.py`, `install.py`, `requirements.txt`. All files in the directory `solver`, `transformer`, `utils`, and `data`, are from the original repository, with slight changes such as disabling output.

### CODE REVIEW

File: torchbenchmark/models/speech_transformer/config.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    checkpoint = False
+    continue_from = False
+    model_path = 'final.pth.tar'
+    print_freq = 10
+    visdom = 0
+    visdom_lr = 0
+    visdom_epoch = 0
+    visdom_id = 0
+    # input files
+    train_json = "input_data/train/data.json"
```
Comment (bitfort) [inline torchbenchmark/models/speech_transformer/config.py:57]: Are these paths relative to torchbench? I'm not a big fan of hardcoded paths in general, but I also don't see a better solution :)

Perhaps at least comment so they are clearly called out.

**Review by bitfort — CHANGES_REQUESTED** [review]
Super impressive you got this together Xu! 

I think it looks really good, I just want to take another look at it after asking a few questions.

To help with the review, I don't want to overly comment on code that was part of the original model since we don't really want to change that. Could you update the PR description briefly which parts are original and which parts you added / modified?

File: torchbenchmark/models/speech_transformer/bin/recognize.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                    help='Max output length. If ==0 (default), it uses a '
+                    'end-detect function to automatically find maximum '
+                    'hypothesis lengths')
+
+
+def recognize(args):
+    model, LFR_m, LFR_n = Transformer.load_model(args.model_path)
+    print(model)
+    model.eval()
+    model.cuda()
```
[the remark this answers is not in the document]
Comment (xuzhao9) [reply torchbenchmark/models/speech_transformer/bin/recognize.py:39]: This file is copied from https://github.com/kaituoxu/Speech-Transformer/blob/master/src/bin/recognize.py, it is actually not used, I will remove this file later.

File: torchbenchmark/models/speech_transformer/bin/recognize.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (xuzhao9) [reply torchbenchmark/models/speech_transformer/bin/recognize.py:1]: This file is copied from https://github.com/kaituoxu/Speech-Transformer/blob/master/src/bin/recognize.py, it is actually not used, I will remove this file later.

File: torchbenchmark/models/speech_transformer/bin/recognize.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (xuzhao9) [reply torchbenchmark/models/speech_transformer/bin/recognize.py:39]: The SpeechTransformer model works only on CUDA, I will add comments in other places to explain.

File: torchbenchmark/models/speech_transformer/config.py (hunk 1 above)
Comment (xuzhao9) [reply torchbenchmark/models/speech_transformer/config.py:57]: These paths are relative to the directory of `__file__`, I will add comments.

**Review by bitfort — APPROVED** [review]
Approving pending any comments, thanks Xu!

**Pull request merged**